tgui/packages/tgui/interfaces/ExodroneConsole.tsx

```diff
@@ -0,0 +1,793 @@
+    travel_error,
+    drone_travel_coefficent,
+    all_bands,
+    drone_status,
+  } = data;
+
+  const travel_cost = target_site => {
+    if (site) {
+      return Math.max(Math.abs(site.distance - target_site.distance), 1) * drone_travel_coefficent;
+    }
+    else {
+      return target_site.distance * drone_travel_coefficent;
+    }
+  };
+  const [
+    choosingTools,
+    setChoosingTools,
+  ] = useLocalState(context, 'choosingTools', false);
+  const [
+    TravelDimmerShown,
+    setTravelDimmerShown,
+  ] = useLocalState(context, 'TravelDimmerShown', false);
+
+  const travel_to = ref => {
+    setTravelDimmerShown(false);
+    act("start_travel", { "target_site": ref });
+  };
+
+  return (
+    drone_status === "travel" && (
+      <TravelDimmer />
+    ) || (
+      <Section
+        title="Travel Destinations"
+        fill
+        scrollable
+        buttons={
+          <>
+            {props.showCancelButton && (
+              <Button
+                ml={5}
+                mr={0}
+                content="Cancel"
+                onClick={() => setTravelDimmerShown(false)} />
+            )}
+            <Box mt={props.showCancelButton && -3.5}>
+              <DroneStatus />
+            </Box>
+          </>
+        }>
+        {((sites && !sites.length) && !choosingTools) && (
+          <NoSiteDimmer />
+        )}
+        {site && (
+          <Section
+            mt={1}
+            title="Home"
+            buttons={
+              <Box>
+                <Button
+                  mr={1}
+                  content={can_travel ? "Launch!" : travel_error}
+                  onClick={() => travel_to(null)}
+                  disabled={!can_travel} />
+                ETA: {formatTime(site.distance * drone_travel_coefficent, "short")}
+              </Box>
+            }
+          />
+        )}
+        {sites.filter(destination => !site || destination.ref !== site.ref).map(destination => (
+          <Section
+            key={destination.ref}
+            title={destination.name}
+            buttons={
+              <>
+                <Button
+                  mr={1}
+                  content={can_travel ? "Launch!" : travel_error}
+                  onClick={() => travel_to(destination.ref)}
+                  disabled={!can_travel} />
+                ETA: {formatTime(travel_cost(destination), "short")}
+              </>
+            }>
+            <LabeledList>
+              <LabeledList.Item label="Location">
+                {destination.coordinates}
+              </LabeledList.Item>
+              <LabeledList.Item label="Description">
+                {destination.description}
+              </LabeledList.Item>
+              <LabeledList.Divider />
+              {Object.keys(all_bands).filter(band => (destination.band_info[band] !== undefined && destination.band_info[band] !== 0)).map(band => (<LabeledList.Item key={band} label={band}>{destination.band_info[band]}</LabeledList.Item>))}
```

Comment:
okay but this filter does genuinely need to be a constant on the function itself. also tab out this map!